src/lib/swapper/swappers/ArbitrumBridgeSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,225 @@
+    // by checking against Arbitrum bridge's own mappings, which uses different sources (Coingecko, Gemini, Uni and its own lists at the time of writing)
+    const arbitrumBridgeErc20L2Address = await (bridger as Erc20Bridger).getL2ERC20Address(
+      erc20L1Address,
+      l1Provider,
+    )
+    const arbitrumBridgeErc20L1Address = await (bridger as Erc20Bridger).getL1ERC20Address(
+      erc20L2Address,
+      l2Provider,
+    )
+
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L1Address), getAddress(erc20L1Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L1 ERC20 address: ${erc20L1Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L2Address), getAddress(erc20L2Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L2 ERC20 address: ${erc20L2Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+  }
+
+  if (sellAsset.assetId === usdcAssetId || sellAsset.assetId === usdcOnArbitrumAssetId) {
+    // https://www.circle.com/en/cross-chain-transfer-protocol
+    throw new Error('cctp not implemented')
+  }
+
+  if (isDeposit && !isEthBridge) {
+    const estimatedParentChainGas = fetchTokenFallbackGasEstimates()
+    const allowanceContract = await (bridger as Erc20Bridger).getL1GatewayAddress(
+      erc20L1Address,
+      l1Provider,
+    )
+
+    const { fast } = await adapter.getGasFeeData()
+
+    return Ok({
+      id: uuid(),
+      receiveAddress,
+      affiliateBps,
+      potentialAffiliateBps,
+      rate: '1',
+      slippageTolerancePercentageDecimal:
+        input.slippageTolerancePercentageDecimal ??
+        getDefaultSlippageDecimalPercentageForSwapper(SwapperName.ArbitrumBridge),
+      steps: [
+        {
+          // https://github.com/OffchainLabs/arbitrum-token-bridge/blob/d17c88ef3eef3f4ffc61a04d34d50406039f045d/packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetailsSteps.tsx#L42
+          // 15 minutes in ms
+          estimatedExecutionTimeMs: 15 * 60 * 1000,
+          allowanceContract,
+          rate: '1',
+          buyAsset,
+          sellAsset,
+          accountNumber,
+          buyAmountBeforeFeesCryptoBaseUnit: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          buyAmountAfterFeesCryptoBaseUnit: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          feeData: {
+            protocolFees: {},
+            networkFeeCryptoBaseUnit: estimatedParentChainGas.times(fast.gasPrice).toString(),
+          },
+          source: SwapperName.ArbitrumBridge,
+        },
+      ] as SingleHopTradeQuoteSteps,
+    })
+  }
+
+  const request = await (isDeposit
```

Comment:
making the same bridge separation comment here. a bit of extra code, but i like the explicit flows for each bridge scenario imo.  can even hoist out into helper funcs since they are used in a handful of places

Reply:
Extracted most of the logic into `fetchArbitrumBridgeSwap` for reuse in `getTradeQuote`, which is now much cleaner! https://github.com/shapeshift/web/pull/6977/commits/ae57872de7371812330b00aebdddcf300aeb4f94